beetmoverscript/src/beetmoverscript/script.py

```diff
@@ -198,11 +200,29 @@ async def push_to_maven(context):
     context.checksums = dict()  # Needed by downstream calls
     context.raw_balrog_manifest = dict()  # Needed by downstream calls
 
+    check_maven_artifact_map(context)
+
     # overwrite artifacts_to_beetmove with the declarative artifacts ones
     context.artifacts_to_beetmove = task.get_upstream_artifacts(context, preserve_full_paths=True)
     await move_beets(context, context.artifacts_to_beetmove, artifact_map=context.task["payload"]["artifactMap"])
 
 
+def check_maven_artifact_map(context):
+    """Check that versions in artifact map to be valid Maven versions"""
+    version = context.task["payload"]["version"]
+    try:
+        MavenVersion.parse(version)
+    except PatternNotMatchedError:
+        raise ScriptWorkerTaskException(f"bad version '{version}'")
+
+    for artifact_dict in context.task["payload"]["artifactMap"]:
+        for dest_dict in artifact_dict["paths"].values():
+            for dest in dest_dict["destinations"]:
+                dest_file = os.path.basename(dest)
+                if version not in dest_file:
+                    raise ScriptWorkerTaskException(f"version in artifact doesn't match expected version '{version}'")
```

Comment:
Nit: Similarly to the previous exception, let's expand the error message here: 

```suggestion
                    raise ScriptWorkerTaskException(f"Cannot find version '{version}' in file name '{dest_file}'. Path under test: {dest}")
```